usb1/__init__.pyi

```diff
@@ -0,0 +1,545 @@
+                   timeout: Timeout = ...) -> None: ...
+    def setBulk(self,
+                endpoint: Endpoint,
+                buffer_or_len: BufferOrLength,
+                callback: Optional[TransferCallback] = ...,
+                user_data: Optional[UserData] = ...,
+                timeout: Timeout = ...) -> None: ...
+    def setInterrupt(self,
+                     endpoint: Endpoint,
+                     buffer_or_len: BufferOrLength,
+                     callback: Optional[TransferCallback] = ...,
+                     user_data: Optional[UserData] = ...,
+                     timeout: Timeout = ...) -> None: ...
+    def setIsochronous(self,
+                       endpoint: Endpoint,
+                       buffer_or_len: BufferOrLength,
+                       callback: Optional[TransferCallback] = ...,
+                       user_data: Optional[UserData] = ...,
+                       timeout: Timeout = ...,
+                       iso_transfer_length_list: Optional[List[int]] = ...) -> None: ...
+    def getType(self) -> TransferType: ...
+    def getEndpoint(self) -> Endpoint: ...
+    def getStatus(self) -> TransferStatus: ...
+    def getActualLength(self) -> int: ...
+    def getBuffer(self) -> Buffer: ...
+    def getUserData(self) -> UserData: ...
+    def setUserData(self, user_data: UserData) -> None: ...
+    def getISOBufferList(self) -> BufferList: ...
+    def getISOSetupList(self) -> List[Dict[str, int]]: ...
+    def iterISO(self) -> None: ...
+    def setBuffer(self,
+                  buffer_or_len: BufferOrLength) -> None: ...
+    def isSubmitted(self) -> bool: ...
+    def submit(self) -> None: ...
+    def cancel(self) -> None: ...
+
+class USBTransferHelper:
+    def __init__(self,
+                 transfer: Optional[USBTransfer] = ...) -> None: ...
+    def submit(self) -> None: ...
+    def cancel(self) -> None: ...
+    def setEventCallback(self,
+                         event: int,
+                         callback: TransferCallback) -> None: ...
+    def setDefaultCallback(self,
+                           callback: TransferCallback) -> None: ...
+    def getEventCallback(self,
+                         event: int,
+                         default: Optional[TransferCallback] = ...) -> Optional[TransferCallback]: ...
+    def __call__(self,
+                 transfer: USBTransfer) -> None: ...
+    def isSubmited(self) -> bool: ...
+
+class USBPollerThread(threading.Thread):
+    daemon: bool = ...
+
+    def __init__(self,
+                 context: USBContext,
+                 poller: USBPoller,
+                 exc_callback: Optional[ErrorCallback] = ...) -> None: ...
+    def stop(self) -> None: ...
+    @staticmethod
+    def exceptionHandler(exc: BaseException) -> None: ...
+    def run(self) -> None: ...
+
+class USBPoller:
+    def __init__(self,
+                 context: USBContext,
+                 poller: USBPoller) -> None: ...
```

Comment:
While `USBPoller` implements the API it expects from `poller`, other types are accepted (and in fact they should be the norm). Here I would expect a protocol rather than such stronger typing.

Maybe something like this (not tested):
```python
PollEventMask = int
PollResult = List[Tuple[FileDescriptor, PollEventMask]]

class PollerType(Protocol):
  def register(self, fd: FileDescriptor, event_flags: PollEventMask) -> None: ...
  def unregister(self, fd: FileDescriptor) -> None: ...
  def poll(timeout: PollTimeout) -> PollResult: ...

# ...

class USBPoller:
    def __init__(self,
                 context: USBContext,
                 poller: PollerType) -> None: ...
```

Edit: And maybe `class USBPoller(PollerType):` if mypy recognises this as an implementation of that protocol, and lets it fill in the other methods, rather than having to duplicate their declaration (not obvious to me from mypy doc).